### PR TITLE
Fix #3071: HiFi - TopicInfo [BLOCKED #3010]

### LINKED ISSUE #3071: HiFi - TopicInfo
https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/45d183d0-fabf-40d9-aca9-164e55336b7e/specs/

### PR DESCRIPTION
**Pull request by anandwana001**
## Explanation

Fixes #3071 

1. Stories you can play - if more than one story
2. Story you can play if only one story

**This PR is blocked on #3010 as that pr has a download feature gating implemented. Once #3010 get's merge, update this pr to make full use of my downloads feature gaiting.** 

UI - https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/45d183d0-fabf-40d9-aca9-164e55336b7e/specs/

<p>

</p>

### TODO
- [ ] wite test cases
- [ ] click Download 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by anandwana001** [comment]
https://github.com/oppia/oppia-android/pull/2986 blocking this PR as #2986 has a lot of change around `topicTab` enum and TopicInfo which also requires here in this PR. 
@rt4914 Could you review #2986 so we can merge it asap.

**Comment by anandwana001** [comment]
@BenHenning Could you please review this PR? Please ignore the failing checks and UI polishing. 
This PR is introducing the Topic preview(When the topic is not downloaded), and clicking on the download will show a download completion screen and the full topic activity screen. 

Once the PR 3010 gets merged, I will use the download feature in this PR.
I am looking for a review and by the time I am working on the below points, the deadline is so close for its PR. 

TODO
1. I had to add TODO's and create issues for connecting to the domain module.
2. I had to fix the failing checks over CI.

File: scripts/assets/test_file_exemptions.textproto [hunk 1]
```diff
@@ -383,6 +383,12 @@ exempted_file_path: "app/src/main/java/org/oppia/android/app/topic/conceptcard/C
 exempted_file_path: "app/src/main/java/org/oppia/android/app/topic/conceptcard/ConceptCardListener.kt"
 exempted_file_path: "app/src/main/java/org/oppia/android/app/topic/conceptcard/ConceptCardViewModel.kt"
 exempted_file_path: "app/src/main/java/org/oppia/android/app/topic/info/TopicInfoFragmentPresenter.kt"
+exempted_file_path: "app/src/main/java/org/oppia/android/app/topic/info/TopicInfoSkillItemViewModel.kt"
+exempted_file_path: "app/src/main/java/org/oppia/android/app/topic/info/TopicInfoStoryItemViewModel.kt"
+exempted_file_path: "app/src/main/java/org/oppia/android/app/topicdownloaded/TopicDownloadedActivity.kt"
```
Comment (anandwana001) [inline scripts/assets/test_file_exemptions.textproto:395]: Add test for TopicDownloadedActivity and TopicDownloadedFragment

1. Test accessibility label for the activity
2. Test if clicking on view will open topic activity with all four tabs

**Comment by BenHenning** [comment]
@anandwana001 sorry I'm a bit confused. What is contained in this PR that's not in #3010? I would expect just one PR to be needed per https://github.com/oppia/oppia-android/pull/3010#issuecomment-892392083 and then a second PR to follow up with the larger batch of changes that would be blocked on #3275.

**Comment by anandwana001** [comment]
@BenHenning This PR has the actual topic download flow, but not the dagger flag which we introduced in [3010](https://github.com/oppia/oppia-android/pull/3010) as we talked about earlier to have a My downloads feature flag.

**Comment by BenHenning** [comment]
@anandwana001 is both this & #3010 needed to unblock Farees or can we just get away with a single smaller PR subset of functionality? Given that we may be submitting code with very basic tests, I want to make sure the surface area is as small as possible.

**Comment by anandwana001** [comment]
Yes, I am creating a new PR with only the required things and will send to you by EOD.

**Comment by anandwana001** [comment]
Closing this PR as some of the part is done in https://github.com/oppia/oppia-android/pull/3643
and this PR can be used as a reference when implementing the remaining topic download UI part.

**Pull request closed without merging**